test/e2e/rollback_byo_test.go

```diff
@@ -0,0 +1,148 @@
+// +build e2e
+
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"testing"
+
+	"knative.dev/pkg/ptr"
+	pkgTest "knative.dev/pkg/test"
+	"knative.dev/pkg/test/logstream"
+	v1 "knative.dev/serving/pkg/apis/serving/v1"
+	"knative.dev/serving/pkg/apis/serving/v1alpha1"
+	. "knative.dev/serving/pkg/testing/v1alpha1"
+	"knative.dev/serving/test"
+	v1a1test "knative.dev/serving/test/v1alpha1"
+)
+
+func TestRollbackBYOName(t *testing.T) {
+	t.Parallel()
+	cancel := logstream.Start(t)
+	defer cancel()
+
+	clients := Setup(t)
+
+	serviceName := test.ObjectNameForTest(t)
+	byoNameOld := serviceName + "-byo-foo"
+	byoNameNew := serviceName + "-byo-foo-new"
+	names := test.ResourceNames{
+		Service: serviceName,
+		Image:   "helloworld",
+	}
+
+	test.CleanupOnInterrupt(func() { test.TearDown(clients, names) })
+	defer test.TearDown(clients, names)
+
+	withTrafficSpecOld := WithInlineRouteSpec(v1alpha1.RouteSpec{
+		Traffic: []v1alpha1.TrafficTarget{{
+			TrafficTarget: v1.TrafficTarget{
+				RevisionName: byoNameOld,
+				Percent:      ptr.Int64(100),
+			},
+		}},
+	})
+	withTrafficSpecNew := WithInlineRouteSpec(v1alpha1.RouteSpec{
+		Traffic: []v1alpha1.TrafficTarget{{
+			TrafficTarget: v1.TrafficTarget{
+				RevisionName: byoNameNew,
+				Percent:      ptr.Int64(100),
+			},
+		}},
+	})
+
+	t.Logf("Creating a new Service with byo config name %q.", byoNameOld)
+	resources, err := v1a1test.CreateRunLatestServiceReady(t, clients, &names, withTrafficSpecOld, func(svc *v1alpha1.Service) {
+		svc.Spec.ConfigurationSpec.Template.ObjectMeta.Name = byoNameOld
+	})
+	if err != nil {
+		t.Fatalf("Failed to create initial Service: %v: %v", names.Service, err)
+	}
+	originalServiceSpec := resources.Service.Spec
+	url := resources.Route.Status.URL.URL()
+	if _, err = pkgTest.WaitForEndpointState(
```

Comment:
Can we remove this? We have other tests that validate that data plane is setup correctly. This test seems like it is concerned about control plane transitions and for debugging problems it would be good to have a narrower focus.

Reply:
Sure. Just updated this PR